src/app/shared/modules/header/components/search/search.harness.ts

```diff
@@ -0,0 +1,69 @@
+import { ComponentHarness } from '@angular/cdk/testing';
+import { MatInputHarness } from '@angular/material/input/testing';
+import { MatSelectHarness } from '@angular/material/select/testing';
+import { MatButtonHarness } from '@angular/material/button/testing';
+import { MatOptionHarness } from '@angular/material/core/testing';
+import { MatAutocompleteHarness } from '@angular/material/autocomplete/testing';
+
+export class SearchHarness extends ComponentHarness {
+  static hostSelector = 'tk-search';
+  protected getLabel = this.locatorFor('mat-label');
+  protected getInput = this.locatorForOptional(MatInputHarness);
+  protected getSelector = this.locatorForOptional(MatSelectHarness);
+  protected getClearBtn = this.locatorForOptional(
+    MatButtonHarness.with({ text: 'refresh' })
+  );
+  protected getAutocomplete = this.locatorForOptional(MatAutocompleteHarness);
+
+  async labelText(): Promise<string> {
+    const label = await this.getLabel();
+    return label.text();
+  }
+
+  async isInput(): Promise<boolean> {
+    const input = await this.getInput();
+    return Boolean(input);
+  }
+
+  async isSelector(): Promise<boolean> {
+    const selector = await this.getSelector();
+    return Boolean(selector);
+  }
+
+  async clearBtn(): Promise<MatButtonHarness | null> {
+    return await this.getClearBtn();
+  }
+
+  async setInputValue(value: string): Promise<void> {
+    const input = await this.getInput();
```

Comment:
input is optional. so it could throw an error. please check if input is truthy before calling and of its methods.
Applies to other which you are locating using locatorForOptional